.circleci/config.yml

```diff
@@ -0,0 +1,89 @@
+version: 2.1
+jobs:
+  setup:
+    docker:
+      - image: circleci/ruby:3.0.1-node
+        environment:
+          NODE_ENV: test
+          RAILS_ENV: test
+          POSTGRES_HOST: localhost
+          POSTGRES_USER: postgres
+          POSTGRES_DB: the_best_app_test
+          POSTGRES_PASSWORD: password
+    steps:
+      - checkout
+
+      - run: gem update --system
+      - run: gem install bundler
+
+      - restore_cache:
+          keys:
+            - bundle-{{ checksum "Gemfile.lock" }}
+      - run: bundle install --path vendor/bundle
+      - save_cache:
+          key: bundle-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+
+      - restore_cache:
+          keys:
+            - yarn-{{ checksum "yarn.lock" }}
+      - run: yarn install --cache-folder ~/.cache/yarn
+      - save_cache:
+          key: yarn-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+
+      - run: bundle exec rails webpacker:compile
+      - save_cache:
+          key: webpack-{{ .Revision }}
+          paths:
+            - /home/circleci/project/public/packs-test/
+  test:
+    docker:
+      - image: circleci/ruby:3.0.1-node
+        environment:
+          DATABASE_URL: postgresql://postgres:secret@localhost:5432
+          POSTGRES_HOST: localhost
+          POSTGRES_USER: postgres
+          POSTGRES_DB: the_best_app_test
+          POSTGRES_PASSWORD: password
+      - image: circleci/postgres:12.7
+        environment:
```

Comment:
Do we really need this?

Reply:
Yes, we found out that those env are need during the call. And env after are also needed as postgres image didn't work without password and user